src/js/node/http.ts

```diff
@@ -1058,6 +1061,7 @@ export class ServerResponse extends Writable {
       var data = this.#firstWrite || "";
       this.#firstWrite = undefined;
       this.#finished = true;
+      this._implicitHeader();
```

Comment:
I suspect this will have a meaningful negative performance impact because it means that the headers are flushed 

Since this is for compatibility with packages that override `writeHead`, instead we could check if `this.writeHead === originalWriteHead` and only then call it

Reply:
Fixed, but there are a few things you need to be aware of. In Node.js, I found that each call to `write_` checks whether the `header` has been initialized and then calls the `_implicitHeader` function to generate `header`. In Bun, this logic is not relied, at least `_implicitHeader` and `writeHead` are not automatically called. I'm not sure if this implementation difference would affect other libraries.




Related code:

- https://github.com/nodejs/node/blob/41f70568a46efceb59bdb00a727b4fe8d858cb2b/lib/_http_server.js#L337-L341
- https://github.com/nodejs/node/blob/41f70568a46efceb59bdb00a727b4fe8d858cb2b/lib/_http_outgoing.js#L914-L920